keps/sig-node/2033-rootless/README.md

```diff
@@ -0,0 +1,797 @@
+
+Rootless mode executes all components inside UserNS to mitigate vulnerabilities of all components, while Node-level UserNS executes only containers inside UserNS.
+
+Rootless mode and Node-level UserNS do not conflict and can be stacked together. (Node-level UserNS inside Rootless UserNS.)
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1: Production cluster
+
+A user is scared of the past vulnerabilities of kubelet/CRI/OCI, and looking for a way to
+mitigate such potential vulnerabilities.
+
+So the user would want this KEP to be implemented.
+
+The user may face difficulties for deploying stateful workloads because block-based and NFS-based persistent volumes
+mostly do not work (see [Notes/Constraints/Caveats](#notesconstraintscaveats)), but this is not a huge deal,
+when the user can use managed object storages such as Amazon S3, or managed RDBs such as Amazon RDS for storing persistent data.
+
+If the user really needs to run an application that requires the root privileges, the user
+would create a mixed cluster composed of rootful nodes and rootless nodes, and set
+the node selector to ensure the privileged pods to be scheduled on rootful nodes.
+
+#### Story 2: HPC cluster
+
+A user wants to deploy a Kubernetes cluster using shared HPC machines to run scientific research workloads.
+
+However, the machine administrator does not want to allow the user to gain the root privileges,
+because the admin thinks that the user may accidentally break other users' environments.
+
+And yet, the admin hesitates to deploy a shared Kubernetes cluster and to create RBAC-restricted
+accounts for users, because user management in Kubernetes is very difficult.
+
+The user would want this KEP to be implemented so that he/she can deploy Kubernetes without
+convincing the admin.
+
+#### Story 3: `kind` with Rootless Docker/Podman
+
+A user wants to run a test cluster inside Docker/Podman on his/her laptop using `kind`.
+
+However, the user doesn't want Kubernetes/kind/Docker/Podman to gain the root privileges
+because these components may accidentally break the host environment,
+e.g. Docker may modify the host iptables in an unexpected way and break the user's VPN connectivity.
+
+The user would want this KEP to be implemented so that he/she could run `kind` with Rootless Docker/Podman,
+which won't break the host.
```

Comment:
Ack

Reply:
@AkihiroSuda please add this story

```
#### Story 4: control-plane in kubeadm as non-root

[KEP-2568: Run control-plane as non-root in kubeadm](https://github.com/kubernetes/enhancements/tree/master/keps/sig-cluster-lifecycle/kubeadm/2568-kubeadm-non-root-control-plane) is starting down the path of is it possible to avoid common container escapes from control plane components. This spotlights the need for Rootless mode feature especially when these are temporarily created and used to bootstrap an actual cluster in a cloud provider (say using cluster-api)

```

Reply:
I'm confused. The linked [KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-cluster-lifecycle/kubeadm/2568-kubeadm-non-root-control-plane) says `Run node components as non-root in kubeadm` is a non-goal.

Reply:
Maybe, can we call it `#### Story 4: temporary initial cluster for bootstrapping actual cluster`?

Reply:
@AkihiroSuda sounds good.

Reply:
```md
#### Story 4: Temporary initial cluster for bootstrapping

A user needs a temporary initial cluster to bootstrap an actual cluster with Cluster API.

The user wants to avoid having the root privileges.
```

Reply:
just a note because I miss the context of the conversation, kind runs kubeadm under the hood, don't know how this applies here